src/main/java/bio/terra/service/resourcemanagement/DataLocationSelector.java

```diff
@@ -1,19 +1,75 @@
 package bio.terra.service.resourcemanagement;
 
+import bio.terra.buffer.model.HandoutRequestBody;
+import bio.terra.buffer.model.ResourceInfo;
 import bio.terra.model.BillingProfileModel;
 import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetBucketDao;
 import bio.terra.service.resourcemanagement.exception.GoogleResourceException;
 import bio.terra.service.resourcemanagement.exception.GoogleResourceNamingException;
+import bio.terra.service.resourcemanagement.google.GoogleResourceConfiguration;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
 
-public interface DataLocationSelector {
+@Component
+@Profile({"terra", "google"})
+public class DataLocationSelector {
```

Comment:
Does this feel like this class is incorrectly named now?  This is really the project "creator" now no?